apps/backend/src/app/common/interceptors/not-found.interceptor.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  BadRequestException,
+  CallHandler,
+  ExecutionContext,
+  Injectable,
+  InternalServerErrorException,
+  NestInterceptor,
+  NotFoundException,
+} from '@nestjs/common';
+import { catchError, Observable, throwError } from 'rxjs';
+import { EntityNotFoundError } from 'typeorm';
+
+@Injectable()
+export class NotFoundInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+    return next.handle().pipe(
+      catchError((err) => {
+        console.error(err);
+
+        if (err instanceof EntityNotFoundError) {
+          return throwError(() => new NotFoundException());
+        }
+
+        if (err instanceof BadRequestException) {
+          return throwError(() => err);
+        }
+
+        return throwError(() => new InternalServerErrorException());
```

Comment:
Maybe it's better to do not handle unspecified errors and allow to handle it internally by NestJS? That's why you are able to use `next` parameter.

Reply:
Right, updated, please check if it's ok now :)